### ISSUE TITLE
nip-42: successful AUTH must be acknowledged with OK

### ISSUE THREAD

**Issue by Sjors** [scsibug/nostr-rs-relay#213]
In noStrudel I can see in the web socket log that it sends out an `AUTH` message. In the relay log I see `client is authenticated`. But there's no OK message.

[NIP-42](https://github.com/nostr-protocol/nips/blob/master/42.md) says:

> `AUTH` messages sent by clients MUST be answered with an `OK` message, like any `EVENT` message.

A practical consequence is that noStrudel shows an endless spinner:

<img width="656" alt="Scherm­afbeelding 2024-09-16 om 14 08 40" src="https://github.com/user-attachments/assets/e109211e-72fb-40eb-95e5-8e424826a9ed">

**Comment by kumulynja** [comment]
@thesimplekid nip47 aren't accepted/denied with an OK message either, so I guess this issue might be for all ephemeral event types and can be solved generally for all ephemeral events.

**Comment by pats2sats** [comment]
Hi @scsibug, this looks fixed on `master` by `30d0712` (`fix: confirm auth event`), but the latest published tag/image still appears to be `0.9.0`, which predates the fix.

Would you be willing to cut a new tagged release, e.g. `0.9.1` or `0.10.0`, from current `master`?

This matters for clients that correctly wait for the NIP-42 `OK` after `AUTH`; with `0.9.0` they can hang/spin even though the relay logs the client as authenticated. We are currently pinning/building from `64cfcaf` to pick up the fix, but a release tag would make this much safer for relay operators.

For what it's worth, I checked GitHub/sourcehut tags, Docker Hub, crates.io/docs.rs, and public GHCR package metadata; I did not see a newer official packaged release elsewhere.

**Comment by scsibug** [comment]
Yes, agreed, I will publish a new release in the next few days, thank you!

**Comment by scsibug** [comment]
0.10.0 tagged, and released on docker hub.
